### DISCUSSION TITLE
Error :Device requires a Thread Border router" While pairing with Google Home sample app

### DISCUSSION THREAD

**Discussion by Rajesh0071998** [openthread/openthread#9400] [Q&A] [answered]
``I am using the Google Home sample app for the pairing of the matter light using nrf52840DK and my boarder router is rpi3 with nrf52840 Dongle.

As i have tried with android chip-tool it works fine but with Google Home sample app It is shows the error as: "Your Device requires a Thread Border router".
I have flashed matter light Bulb into nrf52840DK and Also Created the device on Home Developer console with the VID=0XFFF1 and PID=0x8005.

Also adding the Logs generated by nrf52840DK Matter light.

`I: BLE advertising is already enabled
I: BLE advertising is already enabled
I: 12679 [DL]BLE connection established (ConnId: 0x00)
I: 12684 [DL]Current number of connections: 1/1
I: 12688 [DL]CHIPoBLE advertising stopped
I: 12692 [DL]NFC Tag emulation stopped
D: 13704 [DL]Write request received for CHIPoBLE RX characteristic (ConnId 0x00)
I: 13711 [BLE]local and remote recv window sizes = 5
I: 13716 [BLE]selected BTP version 4
I: 13719 [BLE]using BTP fragment sizes rx 128 / tx 128.
D: 13734 [DL]ConnId: 0x00, New CCCD value: 0x0002
D: 13738 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 6)
D: 13746 [IN]BLE EndPoint 0x200123a0 Connection Complete
I: 13751 [DL]CHIPoBLE connection established (ConnId: 0x00, GATT MTU: 131)
D: 13787 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 13934 [DL]Write request received for CHIPoBLE RX characteristic (ConnId 0x00)
I: 13941 [EM]>>> [E:60505r S:0 M:221366137] (U) Msg RX from 0:4A2759AA4EDE3884 [0000] --- Type 0000:20 (SecureChannel:PBKDFParamRequest)
I: 13954 [EM]<<< [E:60505r S:0 M:170582553] (U) Msg TX to 0:0000000000000000 [0000] --- Type 0000:21 (SecureChannel:PBKDFParamResponse)
I: 13966 [IN](U) Sending msg 170582553 to IP address 'BLE'
D: 13971 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 128)
D: 14128 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 14136 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 14)
D: 14226 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 14234 [DL]Write request received for CHIPoBLE RX characteristic (ConnId 0x00)
I: 14242 [EM]>>> [E:60505r S:0 M:221366138] (U) Msg RX from 0:4A2759AA4EDE3884 [0000] --- Type 0000:22 (SecureChannel:PASE_Pake1)
I: 15957 [EM]<<< [E:60505r S:0 M:170582554] (U) Msg TX to 0:0000000000000000 [0000] --- Type 0000:23 (SecureChannel:PASE_Pake2)
I: 15968 [IN](U) Sending msg 170582554 to IP address 'BLE'
D: 15973 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 128)
E: 15981 [DL]Long dispatch time: 1747 ms, for event type 16388
D: 16102 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 16110 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 6)
D: 16193 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 16200 [DL]Write request received for CHIPoBLE RX characteristic (ConnId 0x00)
I: 16208 [EM]>>> [E:60505r S:0 M:221366139] (U) Msg RX from 0:4A2759AA4EDE3884 [0000] --- Type 0000:24 (SecureChannel:PASE_Pake3)
I: 16220 [EM]<<< [E:60505r S:0 M:170582555] (U) Msg TX to 0:0000000000000000 [0000] --- Type 0000:40 (SecureChannel:StatusReport)
I: 16232 [IN](U) Sending msg 170582555 to IP address 'BLE'
D: 16237 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 35)
I: 16246 [SC]SecureSession[0x20003320]: Moving from state 'kEstablishing' --> 'kActive'
D: 16253 [IN]SecureSession[0x20003320]: Activated - Type:1 LSID:45095
D: 16260 [IN]New secure session activated for device <FFFFFFFB00000000, 0>, LSID:45095 PSID:39889!
I: 16268 [SVR]Commissioning completed session establishment step
I: 16275 [DIS]Updating services using commissioning mode 0
E: 16280 [DIS]Failed to remove advertised services: 3
D: 16285 [DL]Using Thread extended MAC for hostname.
I: 16291 [DIS]Advertise commission parameter vendorID=65521 productID=32773 discriminator=3840/15 cm=0
E: 16300 [DIS]Failed to advertise extended commissionable node: 3
D: 16306 [DIS]Scheduling extended discovery timeout in 900s
E: 16311 [DIS]Failed to finalize service update: 3
D: 16316 [DL]CHIPoBLE advertising set to off
I: 16320 [SVR]Device completed Rendezvous process
E: 16324 [DL]Long dispatch time: 124 ms, for event type 16388
D: 16330 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 16372 [DL]Write request received for CHIPoBLE RX characteristic (ConnId 0x00)
I: 16380 [EM]>>> [E:60506r S:45095 M:173501346] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:02 (IM:ReadRequest)
D: 16391 [IM]Received Read request
I: 16396 [EM]<<< [E:60506r S:45095 M:17933421] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:05 (IM:ReportData)
I: 16407 [IN](S) Sending msg 17933421 on secure session with LSID: 45095
D: 16413 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 72)
D: 16508 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 16515 [DL]Write request received for CHIPoBLE RX characteristic (ConnId 0x00)
I: 16523 [EM]>>> [E:60507r S:45095 M:173501347] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:02 (IM:ReadRequest)
D: 16534 [IM]Received Read request
I: 16539 [EM]<<< [E:60507r S:45095 M:17933422] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:05 (IM:ReportData)
I: 16550 [IN](S) Sending msg 17933422 on secure session with LSID: 45095
D: 16557 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 72)
D: 16643 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 16650 [DL]Write request received for CHIPoBLE RX characteristic (ConnId 0x00)
I: 16658 [EM]>>> [E:60508r S:45095 M:173501348] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:02 (IM:ReadRequest)
D: 16669 [IM]Received Read request
I: 16674 [EM]<<< [E:60508r S:45095 M:17933423] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:05 (IM:ReportData)
I: 16685 [IN](S) Sending msg 17933423 on secure session with LSID: 45095
D: 16692 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 71)
D: 16778 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 16785 [DL]Write request received for CHIPoBLE RX characteristic (ConnId 0x00)
I: 16793 [EM]>>> [E:60509r S:45095 M:173501349] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:02 (IM:ReadRequest)
D: 16804 [IM]Received Read request
I: 16809 [EM]<<< [E:60509r S:45095 M:17933424] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:05 (IM:ReportData)
I: 16820 [IN](S) Sending msg 17933424 on secure session with LSID: 45095
D: 16827 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 71)
D: 16913 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 16920 [DL]Write request received for CHIPoBLE RX characteristic (ConnId 0x00)
I: 16928 [EM]>>> [E:60510r S:45095 M:173501350] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:02 (IM:ReadRequest)
D: 16939 [IM]Received Read request
I: 16944 [EM]<<< [E:60510r S:45095 M:17933425] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:05 (IM:ReportData)
I: 16955 [IN](S) Sending msg 17933425 on secure session with LSID: 45095
D: 16962 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 71)
D: 17048 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 17055 [DL]Write request received for CHIPoBLE RX characteristic (ConnId 0x00)
I: 17063 [EM]>>> [E:60511r S:45095 M:173501351] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:02 (IM:ReadRequest)
D: 17074 [IM]Received Read request
I: 17079 [EM]<<< [E:60511r S:45095 M:17933426] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:05 (IM:ReportData)
I: 17090 [IN](S) Sending msg 17933426 on secure session with LSID: 45095
D: 17097 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 97)
D: 17183 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 17190 [DL]Write request received for CHIPoBLE RX characteristic (ConnId 0x00)
I: 17198 [EM]>>> [E:60512r S:45095 M:173501352] (S) Msg RX from 0:FFFFFFFB00000000 [0000] --- Type 0001:02 (IM:ReadRequest)
D: 17209 [IM]Received Read request
I: 17214 [EM]<<< [E:60512r S:45095 M:17933427] (S) Msg TX to 0:FFFFFFFB00000000 [0000] --- Type 0001:05 (IM:ReportData)
I: 17225 [IN](S) Sending msg 17933427 on secure session with LSID: 45095
D: 17232 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 72)
D: 17317 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 19837 [DL]Write request received for CHIPoBLE RX characteristic (ConnId 0x00)
D: 22345 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 3)
D: 22447 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
D: 24967 [DL]Write request received for CHIPoBLE RX characteristic (ConnId 0x00)
D: 27475 [DL]Sending indication for CHIPoBLE TX characteristic (ConnId 00, len 3)
D: 27532 [DL]Indication for CHIPoBLE TX characteristic done (ConnId 0x00, result 0x00)
I: 28568 [DL]BLE GAP connection terminated (reason 0x13)
D: 28573 [IN]Clearing BLE pending packets.
I: 28577 [DL]Current number of connections: 0/1

`

How to resolve this issue?

**Comment by jwhui** [comment, marked as answer]
The error is likely caused by the lack of Thread Credentials stored on the Android device.

A Matter Commissioner requires knowledge of the Thread Credentials when provisioning a Matter-over-Thread device.

The Google Home App on Android retrieves Thread Credential information from the [Thread Play Services APIs](https://developers.home.google.com/matter/thread).

Make sure you have the Thread Credentials stored in Thread Play Services.
